test/MUXControlsTestApp/VisualTreeTestBase.cs

```diff
@@ -0,0 +1,360 @@
+            TestCaseName = elements[elements.Length - 1];
+            MasterFileNamePrefix = elements[elements.Length - 2] + "_" + TestCaseName;
+            VisualTreeDumpFilter = null;
+            PropertyValueTranslator = null;
+            LogTestContext();
+            ShouldLogMasterFile = TestContextContainsKey(LogMasterFileRuntimeParameterName);
+
+            OnTestInitialized();
+        }
+        protected virtual void OnTestInitialized() { }
+
+        private void LogTestContext()
+        {
+#if DEBUG
+            Log.Comment("FullyQualifiedTestClassName: " + TestContext.FullyQualifiedTestClassName);
+            Log.Comment("TestName: " + TestContext.TestName);
+#endif
+            Log.Comment("MasterFileNamePrefix: " + MasterFileNamePrefix);
+
+            if (TestContextContainsKey(LogMasterFileRuntimeParameterName))
+            {
+                Log.Comment(LogMasterFileRuntimeParameterName + ": True");
+            }
+            else
+            {
+                Log.Comment(LogMasterFileRuntimeParameterName + ": False");
+            }
+        }
+
+
+        private void VisualTreeCompare(UIElement root, string masterFilePrefix, string theme)
+        {
+            string content = "";
+            string expectedContent = "";
+
+            RunOnUIThread.Execute(() =>
+            {
+                content = VisualTreeDumper.DumpToXML(root, PropertyValueTranslator, VisualTreeDumpFilter);
+            });
+
+            MasterFileStorage storage = new MasterFileStorage(!ShouldLogMasterFile, masterFilePrefix, theme);
+            string bestMatchedMasterFileName = storage.BestMatchedMasterFileName;
+            string expectedMasterFileName = storage.ExpectedMasterFileName;
+
+            Log.Comment("Target master file: " + expectedMasterFileName);
+            Log.Comment("Best matched master file: " + bestMatchedMasterFileName);
+
+            VisualTreeOutputCompare result = new VisualTreeOutputCompare("", "");
+            if (String.IsNullOrEmpty(bestMatchedMasterFileName))
+            {
+                result.AddError("Can't find master file for " + TestCaseName);
+            }
+            else
+            {
+                expectedContent = MasterFileStorage.GetMasterFileContent(bestMatchedMasterFileName);
+                result = new VisualTreeOutputCompare(content, expectedContent);
+            }
+
+            if (result.HasError())
+            {
+                storage.LogMasterFile(expectedMasterFileName, content);
+                storage.LogMasterFile(expectedMasterFileName + ".orig", expectedContent);
+
+                Log.Comment(result.ToString());
+                string error = String.Format("Compare failed, but {0} is put into {1}", expectedMasterFileName, storage.StorageLocation);
+                Verify.Fail(error);
+            }
+        }
+
+        // Avoid too much logs, and only log message when !flag.
+        private void CheckTrue(bool flag, string message)
```

Comment:
I feel like this should still have "Verify" in the name to make it clear it maps to standard test failures. VerifyTrue()?

Reply:
whole class are Removed